### ISSUE TITLE
Poor or no support for JSDoc var or member tags

### ISSUE THREAD

**Issue by LqdBcnAtWork** [microsoft/vscode#200033]
Type: <b>Bug</b>

Reference JSDoc's documentation:

https://jsdoc.app/tags-member

Issue:
A @member or @var tag at the root level is supposed to document a variable in the global scope. This is not the case. Hovering over the type or variable name reveals nothing. Also no auto completion for the variable.

Reproduce:
Start new JS (or probably TS file too), start writing a JSDoc comment. Write a @member or @var tag. Try to use the documented variable elsewhere. No autocompletion, and TS will also throw an error that the variable doesn't exist [(planning on reporting there too)](https://github.com/microsoft/TypeScript/issues/56674).

VS Code version: Code - Insiders 1.85.0-insider (c46c2f120b96f1d538333b263e42d1f133a200f0, 2023-12-05T09:50:32.134Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 7 7700X 8-Core Processor              (16 x 4500)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.19GB (23.08GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256197
vslsvsres303:30308271
pythontb:30258533
pythonptprofiler:30281269
vshan820:30294714
vscod805cf:30301675
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30404738
py29gd2263:30784851
vscaat:30438846
vsclangdf:30492506
c4g48928:30535728
dsvsc012:30540252
showlangstatbar:30737417
fixshowwlkth:30771523
showindicator:30805243
pythongtdpath:30726887
i26e3531:30792625
welcomedialogc:30812479
pythonnosmt12:30779711
pythonidxpt:30768918
pythonnoceb:30776497
asynctok:30898717
dsvsc013:30777762
dsvsc014:30777825
pythonmpsinfo:30901776
dsvsc015:30821418
pythontestfixt:30866404
pythonregdiag2:30902439
pyreplss1:30879911
pythonmypyd1:30859725
pythoncet0:30859736
pythontbext0:30879054
accentitlementst:30870582
dsvsc016:30879898
dsvsc017:30880771
dsvsc018:30880772
aa_t_chat:30882232

```

</details>

**Comment by mjbvz** [comment]
Duplicate of https://github.com/microsoft/TypeScript/issues/56674

**Comment by LqdBcnAtWork** [comment]
@mjbvz you marked my issue as a duplicate _of my own issue_. Which I made at the _same time as this one_.

Does TS drive the autocompletion for VS Code?

If not, then I really doubt this issue is truly closed. It might want to wait on TS to implement the feature first. But for those who do not use TS or TS type checking for JavaScript with JSDoc. Having auto completion for the `@member` tag to the same extent as `@prop`, `@param`, `@type`, and `@typedef` should still be fixed.

_This_ issue is for VS Code and it's broken implementation of `@member`, TS's 56674 (which is authored by me) is requesting TS to implement support for `@member`. These go together (and as such are linked to each other). They are _not duplicates of each other_.

**Comment by mjbvz** [comment]
TS implements VS Code's JS support too. I would have moved the issue over but you already created an upstream issue
